mirrord/agent/src/sniffer.rs

```diff
@@ -203,147 +234,59 @@ fn get_tcp_packet(eth_packet: Vec<u8>) -> Option<(TcpSessionIdentifier, TcpPacke
-                command: SnifferCommands::NewAgent(sender),
-            })
-            .await?;
-
-        Ok(Self {
-            client_id,
-            sender: sniffer_sender,
-            receiver,
-            task_status,
-        })
-    }
-
-    /// Send the given command to the connected [`TcpConnectionSniffer`].
-    async fn send_command(&mut self, command: SnifferCommands) -> Result<(), AgentError> {
-        let command = SnifferCommand {
-            client_id: self.client_id,
-            command,
-        };
-
-        if self.sender.send(command).await.is_ok() {
-            Ok(())
-        } else {
-            Err(self.task_status.unwrap_err().await)
-        }
-    }
+pub(crate) struct TcpConnectionSniffer {
+    command_rx: Receiver<SnifferCommand>,
+    raw_capture: RawCapture,
 
-    /// Return the next message from the connected [`TcpConnectionSniffer`].
-    pub async fn recv(&mut self) -> Result<DaemonTcp, AgentError> {
-        match self.receiver.recv().await {
-            Some(msg) => Ok(msg),
-            None => Err(self.task_status.unwrap_err().await),
-        }
-    }
+    port_subscriptions: Subscriptions<Port, ClientId>,
+    sessions: TCPSessionMap,
 
-    /// Tansform the given message into a [`SnifferCommands`] and pass it to the connected
-    /// [`TcpConnectionSniffer`].
-    pub async fn handle_client_message(&mut self, message: LayerTcp) -> Result<(), AgentError> {
-        self.send_command(message.into()).await
-    }
+    client_txs: HashMap<ClientId, Sender<SniffedConnection>>,
+    clients_closed: FuturesUnordered<ClientClosed>,
 }
 
-impl Drop for TcpSnifferApi {
-    fn drop(&mut self) {
-        self.sender
-            .try_send(SnifferCommand {
-                client_id: self.client_id,
-                command: SnifferCommands::AgentClosed,
-            })
-            .unwrap();
+impl fmt::Debug for TcpConnectionSniffer {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("TcpConnectionSniffer")
+            .field("clients", &self.client_txs.keys())
+            .field("port_subscriptions", &self.port_subscriptions)
+            .field("open_tcp_sessions", &self.sessions.keys())
+            .finish()
     }
 }
 
-pub(crate) struct TcpConnectionSniffer {
-    port_subscriptions: Subscriptions<Port, ClientId>,
-    receiver: Receiver<SnifferCommand>,
-    client_senders: HashMap<ClientId, Sender<DaemonTcp>>,
-    raw_capture: RawCapture,
-    sessions: TCPSessionMap,
-    //todo: impl drop for index allocator and connection id..
-    connection_id_to_tcp_identifier: HashMap<ConnectionId, TcpSessionIdentifier>,
-    index_allocator: IndexAllocator<ConnectionId, 100>,
-}
-
 impl TcpConnectionSniffer {
     pub const TASK_NAME: &'static str = "Sniffer";
 
     /// Runs the sniffer loop, capturing packets.
-    #[tracing::instrument(level = "trace", skip(self))]
+    #[tracing::instrument(level = Level::DEBUG, skip(cancel_token), err)]
```

Comment:
`debug` intentional?

Reply:
Yup, this way this trace's data will be included in some debug-level logs. I added some debug-level logs for stuff that should not generate lots of spam and kept spam on trace level